obspy/io/gcf/core.py

```diff
@@ -33,68 +142,584 @@ def merge_gcf_stream(st):
+          =================  ==============  ===========================
+          stats.gcf.sysType  stats.gcf.digi  digitizer
+          =================  ==============  ===========================
+          0                  0               unknown (probably DM24 Mk2)
+          1                  0               DM24
+          1                  1               CD24
+          2                  0               Affinity
+          2                  1               Minimus
+          =================  ==============  ===========================
+     ``stats.gcf.ttl`` : int
+         tap-table-lookup to retrieve sequence of decimation filters used (see 
+         `GCF reference<https://www.guralp.com/apps/ok?doc=GCF_format>` 
+         for further info)
+     ``stats.gcf.blk`` : int
+         Lowest block number in gcf file of blocks included in trace (count start at 0)
+     ``stats.gcf.FIC`` : int 
+         Forward Integration Constant (i.e. first data value)
+     ``stats.gcf.RIC`` : int 
+         Reverse Integration Constant (ideally last data value)
+     ``stats.gcf.stat`` : int
+         Status code, available codes are:
+         ==== ==========================================================================
+         code description
+         ==== ==========================================================================
+         -1   not a data block
+         0    no detected block issues
+         1    stream ID use extended system ID format
+         2    stream ID use double-extended system ID format
+         3    unknown compression
+         4    to few/many data samples indicated in block header
+         5    start of first data sample is negative
+         9    failure to decode header values
+         10   failure to decode data values (last data != RIC, may be due to bad RIC!!!)
+         11   first first difference != 0 (may not be critical!!!)
+         21   status codes 10 + 11
+         ==== ==========================================================================
+   
+   ..rubric:: Example
 
     >>> from obspy import read
     >>> st = read("/path/to/20160603_1955n.gcf", format="GCF")
-    """
-    traces = []
-    with open(filename, 'rb') as f:
-        while True:
-            try:
-                if headonly:
-                    header = libgcf.read_header(f, **kwargs)
-                    if header:
-                        traces.append(Trace(header=header))
-                else:
-                    hd = libgcf.read(f, **kwargs)
-                    if hd:
-                        traces.append(Trace(header=hd[0], data=hd[1]))
-            except EOFError:
-                break
-    st = Stream(traces=traces)
-    if headonly:
-        st = merge_gcf_stream(st)
-    else:
-        st.merge(-1)
-    return st
+   """
+   if not os.path.exists(filename):
+      raise IOError("file %s could not be located (erroneous path?)" % (filename))
```

Comment:
Since we are Python 3 only, for new additions f-string notation would be nice, but thats just cosmetics, so feel free to ignore =)

Reply:
Well, f-strings appeared from python 3.6 (or are they backported to all 3.x versions)? In any case unless support for %-formatting are expected to be deprecated in future versions of python I'd prefer to keep them around :-þ